src/components/charts/chartController.vue

```diff
@@ -394,16 +394,34 @@ export default {
     // this is called when there are multiple time period charts, it returns an array of all the
     // chart dates for that index so that they can be displayed on multiple lines
     buildXaxisTick (index) {
+      const interval = this.$store.getters[this.path + '/intervalUnit']
+
       if (this.chartData.datasets) {
         let tick = []
         for (let chart of this.chartData.datasets) {
           let date = ''
 
           if (chart.data[index]) {
             if (chart.data[index].originalX) {
-              date = chart.data[index].originalX.toLocaleDateString('en-US', { month: 'numeric', day: 'numeric' })
+              date = chart.data[index].originalX
             } else {
-              date = chart.data[index].x.toLocaleDateString('en-US', { month: 'numeric', day: 'numeric' })
+              date = chart.data[index].x
+            }
+
+            switch (interval) {
+              case 'day':
+                date = date.toLocaleDateString('en-US', { month: 'numeric', day: '2-digit' })
+                break
+              case 'hour':
+              case 'minute':
+                date =
+                  date.toLocaleDateString('en-US', { month: 'numeric', day: '2-digit' }) +
```

Comment:
Not sure if day of the week or day of the month is better, but if we want to use day of the week on comparison graphs too (same format), this worked: `date.toLocaleDateString('en-US', { weekday: 'short' }).substring(0, 2) +`

See also: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString#using_options